### PR TITLE
Feat/d auth test

### PR DESCRIPTION
**Pull request by dpnova**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] If this change fixes a bug (or a performance problem), is a regression
  test (or a benchmark) included?
- [ ] Is a documentation update included (if this change modifies
  existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### CODE REVIEW

File: web/user_accounts/middlewares.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+
+class Auth0Middleware(object):
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if not re.match("^/graphql.*", request.path):
+            return self.get_response(request)
+        # if request.method == "GET":
```
Comment (dpnova) [inline web/user_accounts/middlewares.py]: Could get rid of these comments

File: web/user_accounts/middlewares.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                status=401
+            )
+
+        # SET USER (if exists, otherwise pass)
+        try:
+            jwt_account = JWTAccount.objects.get(access_token=token)
+            request.user = jwt_account.user
+        except Exception:
+            request.user = AnonymousUser()
+            # return JsonResponse(data={"code": "invalid_user",
```
Comment (dpnova) [inline web/user_accounts/middlewares.py]: These comments too.

File: web/user_accounts/schema.py [hunk 3]
```diff
@@ -13,11 +21,23 @@ class Meta:
...
 
 
+class JWTAccount(DjangoObjectType):
+    """The JWT account for a user."""
+
+    class Meta:
+        model = JWTAccountModel
+
+
+class SmartEquityException(Exception):
```
Comment (dpnova) [inline web/user_accounts/schema.py]: I think this should subclass the exception in `web/utils.py`

File: web/user_accounts/tests/test_schema.py [hunk 4]
```diff
@@ -39,3 +59,259 @@ def test_resolve_profile_signed_out(self, profile):
...
+    ):
+
+        context_mock = Mock()
+        client = Client(graphene.Schema(
+            mutation=Mutation,
+        ))
+
+        executed = client.execute(
+                '''
+                mutation createUpdateJWTUserMutation {
```
Comment (dpnova) [inline web/user_accounts/tests/test_schema.py]: Should we move these test queries to the docs like other mutations? cc: @cballantyne

File: web/user_accounts/middlewares.py (hunk 1 above)
Comment (cballantyne) [reply web/user_accounts/middlewares.py]: Done.

File: web/user_accounts/middlewares.py (hunk 2 above)
Comment (cballantyne) [reply web/user_accounts/middlewares.py]: Done.

File: web/user_accounts/schema.py (hunk 3 above)
Comment (cballantyne) [reply web/user_accounts/schema.py]: Yep. It now subclasses ESGraphQLException.

File: web/se/schemadoc.py [hunk 5]
```diff
@@ -318,12 +318,3 @@ class Docs():
             }
         }
     """
-
-    Profile = """
```
Comment (cballantyne) [inline web/se/schemadoc.py]: query was in user_accounts.schema but docs were in se.schemadoc. Just moved this to user_accounts.schemadoc.

**Pull request merged**